core/src/main/java/com/datastax/oss/driver/internal/core/cql/DefaultBoundStatement.java

```diff
@@ -358,7 +358,7 @@ public ByteBuffer getRoutingKey() {
       if (indices.isEmpty()) {
         return null;
       } else if (indices.size() == 1) {
-        return getBytesUnsafe(indices.get(0));
+        return isSet(0) ? getBytesUnsafe(indices.get(0)) : null;
```

Comment:
probably worth noting that the multi-column partition key index code branch has already fixed this bug at line 367 below

Reply:
this exactly ^.    On initial look, I came to the same conclusion and felt that you could even just delete this branch and fall into the else and the behavior matches this change exactly, but I think this branch sort of exists as an optimization (avoids creating an array).